doc/sphinx/source/fusion/pattern-matching.rst

```diff
@@ -0,0 +1,81 @@
+.. fusion/pattern-matching.rst: 
+
+Applying pattern matcher to fuse ops
+-------------------------------------
+
+Before delving into the details of pattern matching, note that the sole purpose 
+of the pattern-matching step is to **find** patterns in the framework's graph of 
```

Comment:
`the framework's graph of outputs` -> `the nGraph IR`

Reply:
updated